redis/src/push_manager.rs

```
@@ -0,0 +1,254 @@
+                        data: data.clone(),
+                        con_addr: con_addr.clone(),
+                    };
+                    let is_closed = match &subscription.sender {
+                        #[cfg(feature = "aio")]
+                        PushSender::Tokio(tokio_sender) => tokio_sender.send(push_info).is_err(),
+                        PushSender::Standard(std_sender) => {
+                            let result = std_sender.try_send(push_info);
+                            matches!(result, Err(TrySendError::Disconnected(_)))
+                        }
+                    };
+                    if is_closed {
+                        //TODO check if it causes any problem when guard is hold
+                        self.unsubscribe_from_channel(
+                            channel_name.clone(),
+                            Some(subscription.channel_id),
+                        );
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+/// Manages Push messages for both tokio and std channels
+#[derive(Clone, Default)]
+pub struct PushManager {
+    senders: Arc<ArcSwap<HashMap<PushKind, Vec<PushSender>>>>,
+    subscriptions: Arc<SubscriptionHolder>,
+    psubscriptions: Arc<SubscriptionHolder>,
+    ssubscriptions: Arc<SubscriptionHolder>,
+}
+impl PushManager {
+    /// Try to send `PushInfo` to mpsc channel without blocking
+    pub(crate) fn send(&self, pi: PushInfo) {
+        if let Some(senders) = self.senders.load().get(&pi.kind) {
+            let mut indexes_to_remove = vec![];
+            for (i, sender) in senders.iter().enumerate() {
+                let is_closed = match sender {
+                    #[cfg(feature = "aio")]
+                    PushSender::Tokio(tokio_sender) => tokio_sender.send(pi.clone()).is_err(),
+                    PushSender::Standard(std_sender) => {
+                        let result = std_sender.try_send(pi.clone());
```

you're just silently dropping values if the receiver is full?

---

There is no way propagation back to caller of whatever push producer is. Sender can't be used and SyncSender is blocking. If .send() is used instead of .try_send() then it could simply dead lock the connection.
It's possible to use tracing:error!() tho or only using tokio mpsc.

---

IMO using unbounded throughout might be the best solution. But yes, at tracing the dropped message is a good compromise too.